services/functions/linkdex.ts

```diff
@@ -0,0 +1,81 @@
+import { APIGatewayProxyHandlerV2, APIGatewayProxyStructuredResultV2 } from "aws-lambda"
+import { S3Client, GetObjectCommand, HeadObjectCommand, paginateListObjectsV2 } from "@aws-sdk/client-s3"
+import { CarBlockIterator } from '@ipld/car'
+import { LinkIndexer } from 'linkdex'
+
+export const handler: APIGatewayProxyHandlerV2 = async (event) => {
+  const key = event.queryStringParameters?.key ?? ''
+  const bucket = process.env.BUCKET_NAME ?? ''
+  try {
+    return await getLinkdexReport(key, bucket, new S3Client({}))
+  } catch (err: any) {
+    const details = err?.message ?? err
+    return response({ error: { details }}, { statusCode: 500 })
+  }
+}
+
+export async function getLinkdexReport (key: string, bucket: string, s3: S3Client) {
+  if (!key) {
+    return response({ error: { details: 'key query param is required' }}, { statusCode: 400 })
+  }
+  if (!key.startsWith('raw/') || !key.endsWith('.car') || key.split('/').length !== 4) {
```

Comment:
Why restrict to `raw/` prefix and 4 dirs deep? AFAIK uploadsV2 will be `car-cid/car-cid.car`.

Reply:
This initial implementation is focused on the existing upload flow. We can generalise it once it has proved to be useful. We will likely change this thing quite a bit to be useful to uploads v2